packages/trader/src/App/Components/Elements/__tests__/market-countdown-timer.spec.tsx

```diff
@@ -0,0 +1,95 @@
+
+const mock_default_props = {
+    is_main_page: false,
+    setIsTimerLoading: jest.fn(),
+    onMarketOpen: jest.fn(),
+    symbol: 'WLDAUD',
+};
+
+const default_mock_store = {
+    modules: {
+        trade: {
+            active_symbols: [
+                {
+                    allow_forward_starting: 0,
+                    display_name: 'AUD Basket',
+                    display_order: 26,
+                    exchange_is_open: 1,
+                    is_trading_suspended: 0,
+                    market: 'synthetic_index',
+                    market_display_name: 'Derived',
+                    pip: 0.001,
+                    subgroup: 'baskets',
+                    subgroup_display_name: 'Baskets',
+                    submarket: 'forex_basket',
+                    submarket_display_name: 'Forex Basket',
+                    symbol: 'WLDAUD',
+                    symbol_type: 'forex_basket',
+                },
+            ] as ActiveSymbols,
+        },
+    },
+};
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    WS: {
+        tradingTimes: () => ({
+            api_initial_load_error: false,
+            trading_times: {
+                markets: [
+                    {
+                        submarkets: [
+                            {
+                                symbols: [
+                                    { symbol: 'WLDAUD', times: { open: ['08:00:00.153'], close: ['22:00:00.123'] } },
+                                ],
+                            },
+                        ],
+                    },
+                ],
+            } as unknown as NonNullable<DeepRequired<TradingTimesResponse['trading_times']>>,
+        }),
+    },
+}));
+
+describe('<MarketCountdownTimer />', () => {
+    Date.now = jest.fn(() => new Date(Date.UTC(2017, 1, 14)).valueOf());
+
+    const mockMarketCountdownTimer = (
+        mocked_store: TCoreStores,
+        mock_props: React.ComponentProps<typeof MarketCountdownTimer>
+    ) => {
+        return (
+            <TraderProviders store={mocked_store}>
+                <MarketCountdownTimer {...mock_props} />
+            </TraderProviders>
+        );
+    };
+
+    it('should not render component with children if is_main_page is true', () => {
+        const new_props = { ...mock_default_props };
+        new_props.is_main_page = true;
+        const { container } = render(mockMarketCountdownTimer(mockStore(default_mock_store), new_props));
+
+        expect(container).toBeEmptyDOMElement();
+    });
+    it('should render component with children if is_main_page is false', async () => {
+        jest.useFakeTimers();
+        await act(async () => {
+            const { rerender } = render(mockMarketCountdownTimer(mockStore(default_mock_store), mock_default_props));
+            rerender(mockMarketCountdownTimer(mockStore(default_mock_store), mock_default_props));
+        });
+        jest.runOnlyPendingTimers();
+
```

Comment:
This frightening construction is needed for testing some returned values from `WS` inside of `setTimeout` in `useEffect`.

Reply:
@kate-deriv just a question: will it work with `await waitFor(` instead of `await act(async ()`? :)

Reply:
It worked, thank you)